File: src/Entités/Publication.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entités;

import java.sql.Date;

/**
 *
 * @author 21695
 */
public class Publication {
    private int idPub;
   
    private Date datePub;
    private String Path;
    private String description;

    public Publication() {
    }

    public String getDescription() {
        return description;
    }

    

    public void setDescription(String description) {
        this.description = description;
    }
    

    public int getIdPub() {
        return idPub;
    }

    public void setIdPub(int idPub) {
        this.idPub = idPub;
    }

    public Date getDatePub() {
        return datePub;
    }

    public void setDatePub(Date datePub) {
        this.datePub = datePub;
    }

    public String getPath() {
        return Path;
    }

    public void setPath(String Path) {
        this.Path = Path;
    }

    public Publication(int idPub, Date datePub, String Path, String description) {
        this.idPub = idPub;
        this.datePub = datePub;
        this.Path = Path;
        this.description = description;
    }

    public Publication( Date datePub, String Path, String description) {
        this.datePub = datePub;
        this.Path = Path;
        this.description = description;
    }

    
    @Override
    public String toString() {
        return "Publication{" + "idPub=" + idPub + ", datePub=" + datePub + ", Path=" + Path + ", description=" + description + '}';
    }

    

    

    
    

    
    
}

File: Desktop/gestionevenement/src/Entity/Publication.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Publication
 *
 * @ORM\Table(name="publication")
 * @ORM\Entity
 */
class Publication
{
    /**
     * @var int
     *
     * @ORM\Column(name="idPub", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idpub;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="datePub", type="date", nullable=false)
     */
    private $datepub;

    /**
     * @var string
     *
     * @ORM\Column(name="Path", type="string", length=255, nullable=false)
     */
    private $path;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=255, nullable=false)
     */
    private $description;

    public function getIdpub(): ?int
    {
        return $this->idpub;
    }

    public function getDatepub(): ?\DateTimeInterface
    {
        return $this->datepub;
    }

    public function setDatepub(\DateTimeInterface $datepub): self
    {
        $this->datepub = $datepub;

        return $this;
    }

    public function getPath(): ?string
    {
        return $this->path;
    }

    public function setPath(string $path): self
    {
        $this->path = $path;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }


}
